components/google_drive/actions/find-file/find-file.js

```diff
@@ -0,0 +1,34 @@
+const googleDrive = require("../../google_drive.app");
+const { getListFilesOpts } = require("../../utils");
+
+module.exports = {
+  key: "google_drive-find-file",
+  name: "Find File",
+  description: "Search for a specific file by name",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    googleDrive,
+    drive: {
+      propDefinition: [
+        googleDrive,
+        "watchedDrive",
+      ],
+      description:
+        "The drive to use. If not specified, your personal Google Drive will be used. If you are connected with any [Google Shared Drives](https://support.google.com/a/users/answer/9310351), you can select it here.",
+      optional: true,
+    },
+    nameSearchTerm: {
+      propDefinition: [
+        googleDrive,
+        "fileNameSearchTerm",
+      ],
+    },
+  },
+  async run() {
+    const opts = getListFilesOpts(this.drive || undefined, {
+      q: `name contains '${this.nameSearchTerm}'`,
+    });
+    return (await this.googleDrive.listFilesInPage(null, opts)).files;
```

Comment:
Maybe you can use `lodash.get` or `?` before trying to access `.files`?

Reply:
According to the [API docs](https://developers.google.com/drive/api/v3/reference/files/list#response), a successful request to the [Files: list](https://developers.google.com/drive/api/v3/reference/files/list) endpoint should return a response body containing a `files` property. If the response body is not an object, I think we'd prefer that an error is thrown, for debugging, rather than see an `undefined` return value.